tests/Microsoft.Bot.Builder.Runtime.Tests/Authentication/AllowedCallersClaimsValidationTests.cs

```diff
@@ -0,0 +1,124 @@
+
+        private string audienceClaim = Guid.NewGuid().ToString();
+
+        public static IEnumerable<object[]> GetConfigureServicesSucceedsData()
+        {
+            string primaryAppId = Guid.NewGuid().ToString();
+            string secondaryAppId = Guid.NewGuid().ToString();
+
+            // Null allowed callers
+            yield return new object[]
+            {
+                (string)null,
+                (IConfiguration)TestDataGenerator.BuildConfigurationRoot()
+            };
+
+            // Empty allowed callers array
+            yield return new object[]
+            {
+                (string)null,
+                (IConfiguration)TestDataGenerator.BuildConfigurationRoot(new JObject
+                {
+                    { AllowedCallersClaimsValidator.DefaultAllowedCallersKey, JToken.FromObject(new string[0]) }
+                })
+            };
+
+            // Allow any caller
+            yield return new object[]
+            {
+                primaryAppId,
+                (IConfiguration)TestDataGenerator.BuildConfigurationRoot(new JObject
+                {
+                    { AllowedCallersClaimsValidator.DefaultAllowedCallersKey, JToken.FromObject(new string[] { "*" }) }
+                })
+            };
+
+            // Specify allowed caller
+            yield return new object[]
+            {
+                primaryAppId,
+                (IConfiguration)TestDataGenerator.BuildConfigurationRoot(new JObject
+                {
+                    { AllowedCallersClaimsValidator.DefaultAllowedCallersKey, JToken.FromObject(new string[] { $"{primaryAppId}" }) }
+                })
+            };
+
+            // Specify multiple callers
+            yield return new object[]
+            {
+                primaryAppId,
+                (IConfiguration)TestDataGenerator.BuildConfigurationRoot(new JObject
+                {
+                    { AllowedCallersClaimsValidator.DefaultAllowedCallersKey, JToken.FromObject(new string[] { $"{primaryAppId}", $"{secondaryAppId}" }) }
+                })
+            };
+
+            // Blocked caller throws exception
+            yield return new object[]
+            {
+                primaryAppId,
+                (IConfiguration)TestDataGenerator.BuildConfigurationRoot(new JObject
+                {
+                    { AllowedCallersClaimsValidator.DefaultAllowedCallersKey, JToken.FromObject(new string[] { $"{secondaryAppId}" }) }
+                })
+            };
+        }
+
+        [Theory]
+        [MemberData(nameof(GetConfigureServicesSucceedsData))]
+
+        public async Task AcceptAllowedCallersArray(string allowedCallerClaimId, IConfiguration configuration)
+        {
+            var validator = new AllowedCallersClaimsValidator(configuration);
+
+            var allowedCallersList = configuration.GetSection(AllowedCallersClaimsValidator.DefaultAllowedCallersKey).Get<string[]>();
+
+            if (allowedCallersList != null)
```

Comment:
Should we not test calling ValidateClaimsAsync when the settings from IConfiguration return as null?